staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	fq "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing"
+	kubeinformers "k8s.io/client-go/informers"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+)
+
+// initializeConfigController sets up the controller that processes
+// config API objects.
+func (reqMgmt *requestManagementSystem) initializeConfigController(informerFactory kubeinformers.SharedInformerFactory) {
+	reqMgmt.configQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue")
+	fci := informerFactory.Flowcontrol().V1alpha1()
+	pli := fci.PriorityLevelConfigurations()
+	fsi := fci.FlowSchemas()
+	reqMgmt.plInformer = pli.Informer()
+	reqMgmt.plLister = pli.Lister()
+	reqMgmt.fsInformer = fsi.Informer()
+	reqMgmt.fsLister = fsi.Lister()
+	reqMgmt.plInformer.AddEventHandler(reqMgmt)
+	reqMgmt.fsInformer.AddEventHandler(reqMgmt)
+}
+
+// OnAdd handles notification from an informer of object creation
+func (reqMgmt *requestManagementSystem) OnAdd(obj interface{}) {
+	reqMgmt.configQueue.Add(0)
+}
+
+// OnUpdate handles notification from an informer of object update
+func (reqMgmt *requestManagementSystem) OnUpdate(oldObj, newObj interface{}) {
+	reqMgmt.OnAdd(newObj)
+}
+
+// OnDelete handles notification from an informer of object deletion
+func (reqMgmt *requestManagementSystem) OnDelete(obj interface{}) {
+	reqMgmt.OnAdd(obj)
+}
+
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformer.HasSynced, reqMgmt.fsInformer.HasSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	go wait.Until(reqMgmt.runWorker, time.Second, stopCh)
```

Comment:
```suggestion
	wait.Until(reqMgmt.runWorker, time.Second, stopCh)
```
and no need for `<-stopCh` below

Reply:
Yeah, that could be done.  It would be a further divergence from the usual textual pattern for a controller.